elks/arch/i86/kernel/irq.c

```diff
@@ -132,24 +140,23 @@ void INITPROC irq_init(void)
 {
     flag_t flags;
 
-    init_irq();
+    init_irq();  // PIC initialization
+
+	// TODO: no more need to disable the timer
+	// BEFORE the interrupt initialization
+	// as timer vector is kept as is
```

Comment:
Probably true. Perhaps this should be held to separate PR is that rollback risk decreased, based on legacy PC BIOS. Very early BIOS could not properly handle this case, I really don't know. This is same issue as enable_irq in BIOS driver. Both should wait until seperate PR.

Reply:
Again only an idea, no change for now here.